xla/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -5032,26 +5005,28 @@ absl::StatusOr<CudnnGraph> GetCudnnFlashAttentionOperationGraph(
       .set_io_data_type(ioDataType)
       .set_compute_data_type(cudnn_frontend::DataType_t::FLOAT);
 
+  int uid_offset = 0;
```

Comment:
Can you maybe add a small doc explaining what is `uid`, without going to cudnn documentation it's hard to guess. If it's obvious for anyone familiar with cudnn, ok to skip it.

Also maybe

```
auto next_uid = [uid = 0]() mutable -> CuDnnTensorUID {
  return CuDnnTensorUID(uid++);
};
```

which makes it very clear how uid is assigned

Reply:
Documented UIDs and assignment scheme here https://github.com/openxla/xla/pull/15919/files#diff-92794789634f5ed143bb6f39b59c9e8d9c6d15eaf54cb0c6d19859f673dca6a0R32